.gitignore

```diff
@@ -1,3 +1,51 @@
+# Jekyll
```

Comment:
Is this an auto-generated git ignore?

Reply:
The lines above line 17 are custom for this project (and were present before this PR). Everything after line 17 is autogenerated.

I think our Python ignores were actually autogenerated from the same website before; this PR refreshes the Python ignores and adds VSCode, MacOS, and Ruby ignores.